docs/source/reference.adoc

```diff
@@ -360,3 +360,40 @@ because generating a new token is an expensive operation. Get the access token i
 curl -H "Content-Type: application/json" -X POST -d '{"refresh_token":"<REFRESH_TOKEN>"}' https://auth.openshift.io/api/token/refresh
 
 As a result, you will get a JSON with an access token which you can use in your test.
+
+== Token validation
+
+=== Expiry
+Ensure the token hasn't expired by 'inspecting' the JWT and verifying the`"exp": 1537460165` claim.
+
+=== Verify the signature
+This is necessary to verify that the sender of the JWT is who it says it is and to ensure that the message wasn't changed along the way.
+
+===== Public keys endpoint
+The signature has to be verified against the public key of the corresponding private key used to sign the JWT. The list of public keys in use are available at https://auth.openshift.io/api/token/keys?format=pem
```

Comment:
Worth mentioning that by default (https://auth.openshift.io/api/token/keys) will return keys in JSON Web Key (JWK) format. There are plenty of libs which can parse JWK keys.

Reply:
Good idea https://github.com/fabric8-services/fabric8-auth/pull/610/commits/6e2f2586cab5b15c8bcf1675ffb2213f4a5fe452